e2e/support/helpers/e2e-dashboard-helpers.js

```diff
@@ -204,3 +204,37 @@ export function openDashboardMenu() {
 export const dashboardHeader = () => {
   return cy.findByTestId("dashboard-header");
 };
+
+/**
+ *
+ * @param {number} dashboardId
+ * @param {Object} option
+ * @param {number=} option.col
+ * @param {number=} option.row
+ * @param {number=} option.size_x
+ * @param {number=} option.size_y
+ * @param {string} option.text
+ */
+export const createTextCard = (() => {
+  let id = -1;
```

Comment:
This implicit state worries me. Could you make it a regular function with `id` being `-1` by default?

Reply:
One idea behind not allowing ID to be customizable is to not worry users about what ID they want to use for text cards. [This example](https://github.com/metabase/metabase/pull/33337/files/9c4abbb8dddba5f41c418df698d11258d9b08b05#diff-4f75a007a97c20bb18b979728513ea9a5d115f0f675a59967a45058f59120b32L29) shows that we use positive ID for text cards which doesn't reflect the truth while others don't specify any ID at all ([example 1](https://github.com/metabase/metabase/pull/33337/files/9c4abbb8dddba5f41c418df698d11258d9b08b05#diff-cfb3a167f618f4bcc7c0e578d6485d27a884ef0d24609e570160407d6b4e2dd1L434-L435), [example 2](https://github.com/metabase/metabase/pull/33337/files/9c4abbb8dddba5f41c418df698d11258d9b08b05#diff-311e4636da4bf2af91380bec7d26e9f3fe79f46b9568949e4d2ab204e8701cd9L78-L91)).

I feel fairly justified using auto-decrement ID (negative IDs) in this case since it's not usual for an entity to have negative IDs and engineers shouldn't worry about that too much. Otherwise, we need to keep in mind to use negative IDs for text cards everywhere in our tests, and I don't think it's the goal for engineers to have to manage the text card ID manually anyway.

Also, this uses the same approach in one of the util in our unit test https://github.com/metabase/metabase/blob/28e14d08b4e3391fc91492bc8b63ca369287b0bf/frontend/test/__support__/utils.ts#L1-L5

What's your thought about this?

Reply:
`getNextId` works differently. It's an explicit call to generate an id; here we combine creating a new text card with negative ids. Other cards types (e.g. links) could not be easily created with this approach

Reply:
Sounds great. Thanks for the feedback.

Reply:
Fixed in https://github.com/metabase/metabase/pull/33337/commits/2b3eb037fdd4cf581b60e647e0f72e4fbc05ba44